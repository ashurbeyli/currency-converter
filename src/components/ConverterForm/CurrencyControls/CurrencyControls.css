.mirror-col {
    order: 2;
}