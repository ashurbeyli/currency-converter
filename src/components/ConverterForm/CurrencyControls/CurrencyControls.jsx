import React from 'react';
import { Col, Form, FormControl, InputGroup } from 'react-bootstrap';
import PropTypes from "prop-types";
import './CurrencyControls.css';


const propTypes = {
  label: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  selected: PropTypes.string.isRequired,
  inputValue: PropTypes.string,
  onChangeSelect: PropTypes.func.isRequired,
  onChangeInput: PropTypes.func.isRequired
};

const defaultProps = {
  inputValue: ''
};

const CurrencyControls = ({
  label,
  options,
  selected,
  inputValue,
  onChangeSelect,
  onChangeInput
}) => {
  return (
    <Form.Row>
      <Col xs={3} className={label === 'to' ? 'mirror-col' : ''}>
        <Form.Group>
          <Form.Control as="select" onChange={onChangeSelect} value={selected}>
            {options &&
              options.map(currency => {
                return (
                  <option value={currency} key={currency}>
                    {currency}
                  </option>
                );
              })}
          </Form.Control>
        </Form.Group>
      </Col>
      <Col xs={9}>
        <InputGroup>
          <FormControl
            placeholder={label}
            aria-label={label}
            onChange={onChangeInput}
            value={inputValue}
          />
        </InputGroup>
      </Col>
    </Form.Row>
  );
};

CurrencyControls.propTypes = propTypes;
CurrencyControls.defaultProps = defaultProps;
CurrencyControls.DisplayName = 'CurrencyControls';

export default CurrencyControls;
