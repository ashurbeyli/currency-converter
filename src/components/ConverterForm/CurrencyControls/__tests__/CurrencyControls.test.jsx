import React from 'react';
import { shallow } from 'enzyme';
import { FormControl, InputGroup } from 'react-bootstrap';
import CurrencyControls from '../CurrencyControls';

describe('<ExchangeRate />', () => {
  const onChangeSelect = jest.fn();
  const onChangeInput = jest.fn();

  afterEach(() => {
    onChangeInput.mockReset();
    onChangeSelect.mockReset();
  });

  test('snapshot should match', () => {
    const wrapper = shallow(
      <CurrencyControls
        label="From"
        options={['AZN', 'USD', 'EUR']}
        inputValue={15}
        onChangeSelect={onChangeSelect}
        onChangeInput={onChangeInput}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should onChangeSelect be called', async () => {
    const wrapper = shallow(
      <CurrencyControls label="From" onChangeSelect={onChangeSelect} />
    );

    await wrapper
      .find(FormControl)
      .first()
      .props()
      .onChange(1);

    expect(onChangeSelect).toHaveBeenCalledWith(1);
  });

  test('should onChangeInput be called', async () => {
    const wrapper = shallow(
      <CurrencyControls label="From" onChangeInput={onChangeInput} />
    );

    await wrapper
      .find(InputGroup)
      .find(FormControl)
      .props()
      .onChange(1);

    expect(wrapper).toMatchSnapshot();
    expect(onChangeInput).toHaveBeenCalledWith(1);
  });
});
