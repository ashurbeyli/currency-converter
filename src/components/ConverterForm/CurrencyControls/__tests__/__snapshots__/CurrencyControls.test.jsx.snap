// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExchangeRate /> should onChangeInput be called 1`] = `
<FormRow>
  <Col
    xs={3}
  >
    <FormGroup>
      <FormControl
        as="select"
      />
    </FormGroup>
  </Col>
  <Col
    xs={9}
  >
    <InputGroup>
      <FormControl
        aria-label="From"
        onChange={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        placeholder="From"
      />
    </InputGroup>
  </Col>
</FormRow>
`;

exports[`<ExchangeRate /> snapshot should match 1`] = `
<FormRow>
  <Col
    xs={3}
  >
    <FormGroup>
      <FormControl
        as="select"
        onChange={[MockFunction]}
      >
        <option
          key="AZN"
          value="AZN"
        >
          AZN
        </option>
        <option
          key="USD"
          value="USD"
        >
          USD
        </option>
        <option
          key="EUR"
          value="EUR"
        >
          EUR
        </option>
      </FormControl>
    </FormGroup>
  </Col>
  <Col
    xs={9}
  >
    <InputGroup>
      <FormControl
        aria-label="From"
        onChange={[MockFunction]}
        placeholder="From"
        value={15}
      />
    </InputGroup>
  </Col>
</FormRow>
`;
